File: Programmers/[92334] 신고 결과 받기/ONE/92334.java
import java.util.*;

class Solution {
    public int[] solution(String[] id_list, String[] report, int k) {
        int n = id_list.length;
        int[] answer = new int[n];
        Map<String, Set<String>> reportedMap = new HashMap<>();
        Map<String, Integer> countMap = new HashMap<>();

        for (String s : report) {
            StringTokenizer st = new StringTokenizer(s);
            String user = st.nextToken();
            String reported = st.nextToken();

            if(!reportedMap.containsKey(reported))
                reportedMap.put(reported, new HashSet<>());
            reportedMap.get(reported).add(user);
        }

        for (String s : reportedMap.keySet())
            if (reportedMap.get(s).size() >= k)
                for (String user : reportedMap.get(s))
                    countMap.put(user, countMap.getOrDefault(user, 0) + 1);

        for (int i = 0; i < n; i++) {
            if(!countMap.containsKey(id_list[i]))
                continue;
            answer[i] = countMap.get(id_list[i]);
        }
        return answer;
    }
}

File: Programmers/[92334] 신고 결과 받기/JFe/92334.py
from collections import defaultdict
def solution(id_list, report, k):
    answer = []
    log_dict = defaultdict(list)
    reported = defaultdict(int)
    # 딕셔너리 추가
    for r in report:
        a, b = r.split()
        # 동일한 유저에 대한 신고 횟수는 1회로 처리
        if b not in log_dict[a]:
            log_dict[a].append(b)
            reported[b] += 1
    # 결과 메일 처리
    for id in id_list:
        cnt = 0
        for key in log_dict[id]:
            if reported[key] >= k:
                cnt += 1
        answer.append(cnt)
    return answer